### ISSUE TITLE
Refactor MapboxTripNotification class

### ISSUE THREAD

**Issue by RingerJK** [mapbox/mapbox-navigation-android#4163]
Refactoring of `MapboxTripNotification`:
- 2 almost the same views(collapsed and expanded) which might be moved under interface;
- migrate to koltin;
- review visibility/setText methods along with the first point in the list.

**Comment by Guardiola31337** [comment]
@RingerJK 
> migrate to koltin;

What do you mean by `migrate to koltin`?

**Comment by RingerJK** [comment]
> @RingerJK
> 
> > migrate to koltin;
> 
> What do you mean by `migrate to koltin`?

it's already kotlin, my bad

**Comment by Guardiola31337** [comment]
A couple of follow up comments from https://github.com/mapbox/mapbox-navigation-android/pull/4160

- Should we add `STOP SESSION` text? cc @d-prukop @zugaldia 

![test-app_navigation_is_starting_no_stop_session_text](https://user-images.githubusercontent.com/1668582/112167343-c16f1480-8bc6-11eb-8aac-f90783b2870e.png)

The session has technically started and having only the `X` looks confusing to me.

- Testing the `test-app` examples noticed that when disabling the location from the settings and then re-enabling it, it only recovers for `MapboxNavigationActivity` i.e. if you close the example and navigate to another one, keeps saying `Navigation is starting...` even if the location is ON (not sure though if it's a test-app / activity issue) 👀 

![test-app_navigation_is_starting_notification](https://user-images.githubusercontent.com/1668582/112168246-802b3480-8bc7-11eb-9925-475e3b0b6034.gif)

cc @truburt

**Comment by zugaldia** [comment]
> Should we add STOP SESSION text? cc @d-prukop @zugaldia

Yes. Or any other text that clarifies the available action. That's preferable than no context.

**Comment by RingerJK** [comment]
Scenario `start->stop->start navigation` we've got non-consistatant, so I left it with just cross icon.
![](https://im3.ezgif.com/tmp/ezgif-3-5bfecbf66d37.gif)

we use `End Navigation` for Navigation and `Stop session` for Free drive. Should we always use `Stop session` for the above case? (when location is not available or any other issue happens)

>Testing the test-app examples noticed that when disabling the location from the settings and then re-enabling it, it only recovers for MapboxNavigationActivity i.e. if you close the example and navigate to another one, keeps saying Navigation is starting... even if the location is ON (not sure though if it's a test-app / activity issue) 👀  

I think it's because of `ReplayLocationEngine`, got the same in `MapboxTripProgressActivity` and `VoiceActivity`

**Comment by Guardiola31337** [comment]
Noting here that we're not covering the case in which an _Active Guidance_ session is started without location enabled 👀 

![test-app_notification_with_location_disabled_active_guidance](https://user-images.githubusercontent.com/1668582/112367875-daec8b00-8cb0-11eb-9901-7a7bf7dcd9a6.gif)

cc @RingerJK

**Comment by RingerJK** [comment]
> Noting here that we're not covering the case in which an _Active Guidance_ session is started without location enabled 👀

> cc @RingerJK 

I'm not sure how a puck should run with it, because in that case we depend on NN and rely on their updates.

**Comment by zugaldia** [comment]
@RingerJK @cafesilencio Now that https://github.com/mapbox/mapbox-navigation-android/pull/4448 has landed, what else is missing? Is this ticket still a release blocking issue?
